test/unit/org/apache/cassandra/net/HandshakeTest.java

```diff
@@ -141,6 +143,14 @@ public void testSendIncompatibleOldVersion() throws InterruptedException, Execut
         Assert.assertEquals(current_version + 3, result.incompatible().maxMessagingVersion);
     }
 
+    @Test
+    public void testSendCompatibleMinVersionPre40() throws InterruptedException, ExecutionException
+    {
+        Result result = handshake(VERSION_30, VERSION_30, VERSION_3014, VERSION_30, VERSION_3014);
+        Assert.assertEquals(Result.Outcome.RETRY, result.outcome);
+        Assert.assertEquals(VERSION_3014, result.retry().withMessagingVersion);
```

Comment:
the handshake will now bump up from requested version 30 to 3014 (so to avoid the columnFilter bug)